### ISSUE TITLE
Unable to Use YAML Template in Git Submodule

### ISSUE THREAD

**Issue by alkampfergit** [microsoft/azure-pipelines-yaml#261]
Scenario: 
 Repository called BuildScripts with some YAML Templates
 Project repository that has the previous repository as Submodule in builds/BuildScripts
 Project repository has a yaml build that references templates under BuildsScripts folder

As you can see in the following picture, the build failed to start because it does not find the template file referenced in YAML build, because the file is in the SubModule. 

(Is there any option to enable this scenario? Maybe enabling submodule checkout directly in the YAML definition?)

![image](https://user-images.githubusercontent.com/358545/61375202-92a38580-a89e-11e9-9f2d-0565dc25ed82.png)

**Comment by jsblock78** [comment]
I also would have preferred to do this using submodules.

Instead I was able to use the other repository by using a resource reference in my main yaml, as documented here:

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#using-other-repositories

**Note:** I needed to add an "endpoint", which was missing from the documentation.

Any scripts referenced by templates using this method failed.  To work around this, I kept the submodule configured in my 'new' repository and setup a directory symlink in my 'old' repository so the templates could reference scripts at the expected submodule path even when being consumed by a pipeline in the 'old' repository.
